### PR TITLE
document sending metrics to control plane

### PR DESCRIPTION
**Pull request by ydayagi**
docuement the metrics receiver

https://issues.redhat.com/browse/ECOPROJECT-677
Signed-off-by: Yaron Dayagi <ydayagi@redhat.com>

### CODE REVIEW

File: docs/user-guide/device-metrics.md [hunk 1]
```diff
@@ -30,7 +30,7 @@ Sample allow-list `ConfigMap` (mind `metrics_list.yaml` key):
 
 ```yaml
 apiVersion: v1
-kind: ConfigMap
+kind: ConfigMapi
```
Comment (jakub-dzon) [inline docs/user-guide/device-metrics.md:33]: `ConfigMap`

File: docs/user-guide/device-metrics.md [hunk 2]
```diff
@@ -50,4 +50,128 @@ spec:
...
+## Metrics receiver
+### Overview
+The devices can be configured to write the metrics to a remote server. The client in the device uses [Prometheus Remote Write API](https://docs.google.com/document/d/1LPhVRSFkGNSuU1fBd81ulhsCPR4hkSZyyBj1SZ8fWOM/edit#heading=h.p12mxouu8g0h) (see also [Prometheus Integrations](https://prometheus.io/docs/operating/integrations/)). The device writes metrics until all are written. It then waits 5 minutes for more metrics to be collected.
+### Configuration
+The feature is disabled by default. It is configured via EdgeDevice/EdgeDeviceGroup CRs. Example with inline documentation and defaults:
+```yaml
+spec:
+    metrics:
+      receiverConfiguration:
+        caSecretName: receiver-tls # secret containing CA cert. Secret key is 'ca.crt'. optional
```
Comment (jakub-dzon) [inline docs/user-guide/device-metrics.md:63]: `Optional.` - capital letter after a full stop in this and subsequent lines.

File: docs/user-guide/device-metrics.md [hunk 3]
```diff
@@ -50,4 +50,128 @@ spec:
...
+        requestNumSamples: 10000 # maximum number of samples in each request from device to receiver. optional
+        timeoutSeconds: 10 # timeout for requests to receiver. optional
+        url: https://receiver:19291/api/v1/receive # the receiver's URL. used to indicate HTTP/HTTPS. set to empty in order to disable writing to receiver
+```
+### Example receiver
+We prepared an example for deploying a [Thanos](https://thanos.io) receiver.
+Example includes deployment with and without TLS.
+The receiver listens on port `19291` for incoming writes.
+The deployment's POD includes a container that executes a Thanos querier. You can use it for querying the received metrics. It listens on port `9090`.
```
Comment (jakub-dzon) [inline docs/user-guide/device-metrics.md:72]: "pod"

**Review by jakub-dzon — CHANGES_REQUESTED** [review]

File: docs/user-guide/device-metrics.md (hunk 1 above)
Comment (ydayagi) [reply docs/user-guide/device-metrics.md]: done

File: docs/user-guide/device-metrics.md (hunk 2 above)
Comment (ydayagi) [reply docs/user-guide/device-metrics.md]: done

File: docs/user-guide/device-metrics.md (hunk 3 above)
Comment (ydayagi) [reply docs/user-guide/device-metrics.md]: done

**Review by jakub-dzon — APPROVED** [review]

File: docs/user-guide/device-metrics.md [hunk 4]
```diff
@@ -50,4 +50,128 @@ spec:
     system:
       allowList: 
           name: system-allow-list
+```
+## Metrics receiver
+### Overview
+The devices can be configured to write the metrics to a remote server. The client in the device uses [Prometheus Remote Write API](https://docs.google.com/document/d/1LPhVRSFkGNSuU1fBd81ulhsCPR4hkSZyyBj1SZ8fWOM/edit#heading=h.p12mxouu8g0h) (see also [Prometheus Integrations](https://prometheus.io/docs/operating/integrations/)). The device writes metrics until all are written. It then waits 5 minutes for more metrics to be collected.
```
Comment (gciavarrini) [inline docs/user-guide/device-metrics.md:56]: > The device writes metrics until all are written

I'd like to suggest to reword/clarify a bit. e.g. "all are written": what does it means "all"?

File: docs/user-guide/device-metrics.md [hunk 5]
```diff
@@ -50,4 +50,128 @@ spec:
...
+### Configuration
+The feature is disabled by default. It is configured via EdgeDevice/EdgeDeviceGroup CRs. Example with inline documentation and defaults:
+```yaml
+spec:
+    metrics:
+      receiverConfiguration:
+        caSecretName: receiver-tls # secret containing CA cert. Secret key is 'ca.crt'. Optional
+        requestNumSamples: 10000 # maximum number of samples in each request from device to receiver. Optional
+        timeoutSeconds: 10 # timeout for requests to receiver. Optional
+        url: https://receiver:19291/api/v1/receive # the receiver's URL. used to indicate HTTP/HTTPS. set to empty in order to disable writing to receiver
```
Comment (gciavarrini) [inline docs/user-guide/device-metrics.md:66]: s/used/Used/
s/set/Set/

**Review by machacekondra — APPROVED** [review]

File: docs/user-guide/device-metrics.md (hunk 5 above)
Comment (ydayagi) [reply docs/user-guide/device-metrics.md]: done

File: docs/user-guide/device-metrics.md (hunk 4 above)
Comment (ydayagi) [reply docs/user-guide/device-metrics.md]: I changed it. please recheck

**Review by gciavarrini — APPROVED** [review]

**Pull request merged**